src/less/_flag.less

```diff
@@ -39,171 +39,122 @@
   }
 }
 
-//== Functional Colours
 //
+// == Functional Colours ==
 //
 
-.flag-primary {
-  background:   @primary;
-  border-color: @primary;
-  color:rgb(255,255,255);
+// Mixin
+.flag-variant(@background) {
+  color: contrast(@background, #414141, #ffffff, 60%);
+  background-color: @background;
+  border: @background;
   &:after {
-    border-color: darken(@primary, 5%);
+    border-color: darken(@background, 5%);
     border-right-color: transparent;
   }
   &:before {
-    border-color: darken(@primary, 5%);
+    border-color: darken(@background, 5%);
     border-left-color: transparent;
   }
 }
-.flag-secondary {
-  background:   @secondary;
-  border-color: @secondary;
-  color:rgb(255,255,255);
-  &:after {
-    border-color: darken(@secondary, 5%);
-    border-right-color: transparent;
-  }
-  &:before {
-    border-color: darken(@secondary, 5%);
-    border-left-color: transparent;
-  }
-}
-.flag-success {
-  background:   @success;
-  border-color: @success;
-  color:rgb(255,255,255);
-  &:after {
-    border-color: darken(@success, 5%);
-    border-right-color: transparent;
-  }
-  &:before {
-    border-color: darken(@success, 5%);
-    border-left-color: transparent;
-  }
-}
-.flag-info {
-  background:   @info;
-  border-color: @info;
-  color:rgb(255,255,255);
-  &:after {
-    border-color: darken(@info, 5%);
-    border-right-color: transparent;
+
+.component-flag {
+  .flag-variant(@default);
+  &.primary {
+    .flag-variant(@primary);
   }
-  &:before {
-    border-color: darken(@info, 5%);
-    border-left-color: transparent;
+  &.secondary {
+    .flag-variant(@secondary);
   }
-}
-.flag-warning {
-  background:   @warning;
-  border-color: @warning;
-  color:rgb(255,255,255);
-  &:after {
-    border-color: darken(@warning, 5%);
-    border-right-color: transparent;
+  &.success {
+    .flag-variant(@success);
   }
-  &:before {
-    border-color: darken(@warning, 5%);
-    border-left-color: transparent;
+  &.info {
+    .flag-variant(@info);
   }
-}
-.flag-danger {
-  background:   rgb(216,43,38);
-  border-color: rgb(216,43,38);
-  color:rgb(255,255,255);
-  &:after {
-    border-color: darken(@danger, 5%);
-    border-right-color: transparent;
+  &.warning {
+    .flag-variant(@warning);
   }
-  &:before {
-    border-color: darken(@danger, 5%);
-    border-left-color: transparent;
+  &.danger {
+    .flag-variant(@danger);
   }
 }
 
-//== Position
 //
+// == Position ==
 //
 
-// Flag-Right
-.flag-right {
-  position: absolute;
-  top:   1em;
-  right: 0;
-  margin-right: -.5em;
-  -webkit-border-radius:0 2px 2px 0;
-     -moz-border-radius:0 2px 2px 0;
-      -ms-border-radius:0 2px 2px 0;
-       -o-border-radius:0 2px 2px 0;
-          border-radius:0 2px 2px 0;
-  &:after {
-    content: "";
-    display: block;
-    width:  0;
-    height: 0;
+.component-flag {
+
+  // Right
+  &.right {
```

Comment:
A lot of the the stuff in `.left` and `.right` is duplicated. Feels like this could be streamlined with a mixin.